Neos.Utility.Arrays/Classes/ValueAccessor.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+namespace Neos\Utility;
+
+/*
+ * This file is part of the Neos.Utility.Arrays package.
+ *
+ * (c) Contributors of the Neos Project - www.neos.io
+ *
+ * This package is Open Source Software. For the full copyright and license
+ * information, please view the LICENSE file which was distributed with this
+ * source code.
+ */
+
+/**
+ * Type safe accessing of values from nested arrays
+ */
+class ValueAccessor
+{
+    public function __construct(
+        public readonly mixed   $value,
+        public readonly ?string $pathinfo = null
+    ) {
+    }
+
+    private function createTypeError($message): \TypeError
+    {
+        return new \TypeError(get_debug_type($this->value) . ' ' . $message . ($this->pathinfo ? ' in path ' . $this->pathinfo : ''));
+    }
+
+    public function int(): int
+    {
+        if (is_int($this->value)) {
+            return $this->value;
+        }
+        throw $this->createTypeError("is not an integer");
+    }
+
+    public function float(): float
+    {
+        if (is_float($this->value)) {
+            return $this->value;
+        }
+        throw $this->createTypeError("is not a float");
+    }
+
+    public function number(): int|float
+    {
+        if (is_int($this->value) || is_float($this->value)) {
+            return $this->value;
+        }
+        throw $this->createTypeError("is not a number");
+    }
+
+    public function string(): string
+    {
+        if (is_string($this->value)) {
+            return $this->value;
+        }
+        throw $this->createTypeError("is not a string");
+    }
+
+    /**
+     * @return class-string
+     */
+    public function classString(): string
+    {
+        if (is_string($this->value) && class_exists($this->value)) {
+            return $this->value;
+        }
+        throw $this->createTypeError("is not a class-string");
+    }
+
+    public function array(): array
+    {
+        if (is_array($this->value)) {
+            return $this->value;
+        }
+        throw $this->createTypeError("is not an array");
+    }
+
+    /**
+     * @template T
+     * @param class-string<T> $type
+     * @return object&T
+     */
+    public function object(string $type): object
```

Comment:
```suggestion
    /**
     * @template T
     * @param class-string<T> $className
     * @return object&T
     */
    public function instanceOf(string $className): object
```

Sorry that I keep heckling stuff because of naming preferences :sweat_smile:, but as I went though the methods, this was the only one that didn't tell me immediately what it wanted to check. 

I believe that `instanceOf` would meet common expectations better.